src/tools/pr_8799/src/virtcontainers.rs

```diff
@@ -0,0 +1,118 @@
+use anyhow::{anyhow, Result};
+use std::{collections::HashMap, fs, path::Path};
+
+const CPU_POS_STAT: usize = 38;
+const PROC_STAT_SIZE: usize = 52;
+
+// Create a map of tids-vcpus from /proc/pid subdirectories
+pub fn get_ch_vcpu_tids_by_path(proc_path: &str) -> Result<HashMap<u32, u32>> {
+    let mut tids = HashMap::new();
+
+    if proc_path.trim().is_empty() {
+        return Err(anyhow!("Path must be non-empty strings."));
+    }
+
+    let src = match std::fs::canonicalize(proc_path) {
+        Ok(m) => m,
+        Err(_e) => {
+            return Err(anyhow!("Not a valid /proc/pid path: {}", proc_path));
+        }
+    };
```

Comment:
I think this can be simplified to something like:

```suggestion
    let src = std::fs::canonicalize(proc_path)
        .map_err(|e| anyhow("Invalid /proc/pid path {proc_path:?}: {e:?}"))?;
```